cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -649,6 +649,36 @@ Node uninitializedNode(LocalVariable v) { none() }
  */
 predicate localFlowStep(Node nodeFrom, Node nodeTo) { simpleLocalFlowStep(nodeFrom, nodeTo) }
 
+private predicate simpleFieldNodeLocalFlowStep(Node nodeFrom, Node nodeTo) {
+  // flow from the "innermost" field to the load of that field.
+  exists(FieldNode fieldNode, LoadInstruction load |
+    nodeFrom = fieldNode and
+    nodeTo.asInstruction() = load and
+    not exists(fieldNode.getNextNode()) and
+    fieldNode = getFieldNodeForFieldInstruction(load.getSourceAddress())
+  )
+  or
+  // flow from the "outermost" field to the `ChiInstruction`, or `StoreInstruction`
+  // if no `ChiInstruction` exists.
+  exists(FieldNode fieldNode, AddressOperand addressOperand |
+    nodeFrom = fieldNode and
+    not exists(fieldNode.getObjectNode()) and
+    fieldNode.getNextNode*() = getFieldNodeForFieldInstruction(addressOperand.getDef()) and
```

Comment:
How's the performance of this `*` on kamailio?

Reply:
It ran fine on Kamailio. It doesn't stand out in the clause timing report for any of the stages.